src/images/avatar.ts

```diff
@@ -8,6 +9,36 @@ import {Component, Input, ChangeDetectionStrategy} from 'angular2/core';
 export class NglAvatar {
   @Input() src: string = '';
   @Input() alt: string = '';
-  @Input() size: 'x-small' | 'small' | 'medium' | 'large' = 'medium';
-  @Input() type: 'rectangle' | 'circle' = 'rectangle';
+  @Input() size: 'x-small' | 'small' | 'medium' | 'large';
+
+  @Input('size') set setSize(value: 'x-small' | 'small' | 'medium' | 'large') {
+    this.updateClass(this._size, value);
+    this._size = value;
+  }
+
+  @Input('type') set setType(value: 'rectangle' | 'circle') {
+    this.updateClass(this._type, value);
+    this._type = value;
+  }
+
+  private _type: 'rectangle' | 'circle';
+  private _size: 'x-small' | 'small' | 'medium' | 'large';
+
+  constructor(public element: ElementRef, public renderer: Renderer) {
+    renderer.setElementClass(element.nativeElement, 'slds-avatar', true);
+  }
+
+  ngOnInit() {
+    if (!this._type) {
+      this.renderer.setElementClass(this.element.nativeElement, 'slds-avatar--rectangle', true);
+    }
+
+    if (!this._size) {
+      this.renderer.setElementClass(this.element.nativeElement, 'slds-avatar--medium', true);
+    }
+  }
+
+  private updateClass(oldValue: string, newValue: string) {
+    replaceClass(this, `slds-avatar--${oldValue}`, newValue ? `slds-avatar--${newValue}` : '');
+  }
 };
```

Comment:
If I understand the cases we want to cover here correctly, I think we could make the implementation a bit more compact as such:

```typescript
import {Component, Input, ChangeDetectionStrategy} from 'angular2/core';

@Component({
  selector: 'ngl-avatar',
  changeDetection: ChangeDetectionStrategy.OnPush,
  templateUrl: './avatar.jade',
  host: {
    '[class]': '"slds-avatar slds-avatar--" + type + " slds-avatar--" + size',
  },
})
export class NglAvatar {
  @Input() src: string = '';
  @Input() alt: string = '';
  @Input() type: 'rectangle' | 'circle' = 'rectangle';
  @Input() size: 'x-small' | 'small' | 'medium' | 'large' = 'medium';
};
```

The host binding here is not very pretty but as long as the component remains simple enough, I believe this suffices to express what we want.